### PR TITLE
Multilingual and dynamic colors in manifest.json

### LINKED ISSUE #4449: `manifest.json` is not adequate for splash screens in Chrome
The colors are static and the icon is still based on 256px, despite larger icons up to 512px are supported.

See https://web.dev/add-manifest/#splash-screen

### PR DESCRIPTION
**Pull request by Cyperghost**
Closes #4449

### CODE REVIEW

File: wcfsetup/install/files/lib/data/style/StyleAction.class.php [hunk 1]
```diff
@@ -371,8 +372,9 @@ protected function updateFavicons(Style $style)
                 if (\file_exists($style->getAssetPath() . "favicon.ico")) {
                     \unlink($style->getAssetPath() . "favicon.ico");
                 }
-
-                \unlink($style->getAssetPath() . "manifest.json");
+                foreach (\glob($style->getAssetPath() . "manifest*.json") as $filename) {
```
Comment (dtdesign) [inline wcfsetup/install/files/lib/data/style/StyleAction.class.php:375]:
```suggestion
                foreach (\glob($style->getAssetPath() . "manifest-*.json") as $filename) {
```

Wildcard patterns always have the risk of matching unrelated files and the dash is part of the expected file name.

File: wcfsetup/install/files/lib/system/style/command/CreateManifest.class.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since       6.1
+ */
+final class CreateManifest
+{
+    private readonly int $styleID;
+
+    public function __construct(Style $style)
+    {
+        $this->styleID = $style->styleID;
```
Comment (dtdesign) [inline wcfsetup/install/files/lib/system/style/command/CreateManifest.class.php:26]: There is no need to fetch the style again, just store the passed in Style object.

File: wcfsetup/install/files/lib/system/style/command/CreateManifest.class.php [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        $style->loadVariables();
+        $headerColor = $style->getVariable('wcfHeaderBackground', true);
+        $backgroundColor = $style->getVariable('wcfContentBackground', true);
+        $landingPage = PageCache::getInstance()->getLandingPage();
+
+        $icons = [];
+        // If no favicon is set, use the default favicon,
+        // which comes in 192x192px and 256x256px and starts with `default.`.
+        // These images are located in the `images/favicon/` directory.
+        foreach ($style->hasFavicon ? [192, 256, 512] : [192, 256] as $iconSize) {
```
Comment (dtdesign) [inline wcfsetup/install/files/lib/system/style/command/CreateManifest.class.php:41]: I don’t like this, we should provide a proper 512 px version.

File: wcfsetup/install/files/lib/system/style/command/CreateManifest.class.php [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                $manifest = <<<MANIFEST
+{
+    "name": {$title},
+    "start_url": {$startUrl},
+    "icons": {$icons},
+    "theme_color": "{$headerColor}",
+    "background_color": "{$backgroundColor}",
+    "display": "standalone"
+}
+MANIFEST;
```
Comment (dtdesign) [inline wcfsetup/install/files/lib/system/style/command/CreateManifest.class.php:74]: You should properly indent this. PHP will trim to the left based on the position of `MANIFEST;`. Thus if there are 12 spaces (or whatever) in front of it, the same amount is trimmed from all other lines too.

File: wcfsetup/install/files/lib/data/style/Style.class.php [hunk 5]
```diff
@@ -294,7 +294,9 @@ public function getFaviconAppleTouchIcon()
      */
     public function getFaviconManifest()
     {
-        return $this->getFaviconPath('manifest.json');
+        return WCF::getPath() .
+            FileUtil::getRelativePath(WCF_DIR, $this->getAssetPath()) .
+            'manifest-' . WCF::getLanguage()->languageID . '.json';
```
Comment (dtdesign) [inline wcfsetup/install/files/lib/data/style/Style.class.php:299]: Please use `\sprintf()` instead.

**Review by dtdesign — CHANGES_REQUESTED** [review]

**Review by Fighter456 — CHANGES_REQUESTED** [review]
Small typo.

**Pull request merged**